### PR TITLE
Switch JDK17 to GCC10 as well as JDK18+

### PR DESCRIPTION
**Pull request by sxa**
We enabled GCC10 for JDK18+ at https://github.com/adoptium/temurin-build/pull/2824.

This PR switches JDK17 as well as per [the original request](https://github.com/adoptium/temurin-build/pull/2824) and keeps us in line with Oracle (See mailing list) We held back on changing this for 17 previously because we were too close to the start of the January release cycle.

Adding this in ASAP will allow us to verify that there are no regressions well in advance of the April release of 17.0.3

### CODE REVIEW

File: build-farm/platform-specific-configurations/linux.sh [hunk 1]
```diff
@@ -68,6 +68,9 @@ then
     then
       export CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM} --enable-cuda --with-cuda=$CUDA_HOME"
     fi
+  elif [ "${ARCHITECTURE}" == "aarch64" ] && [ -f /usr/local/cuda-10.2/include/cuda.h ]; then
+    echo PROBABLE JETSON NANO DEVELOPMENT ENVIRONMENT - Enabling CUDA ...
+    export CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM} --enable-cuda --with-cuda=/usr/local/cuda-10.2"
```
Comment (keithc-ca) [inline build-farm/platform-specific-configurations/linux.sh:73]: This seems unrelated to the title of this PR.

Comment (sxa) [reply build-farm/platform-specific-configurations/linux.sh]: Fixed - I'd incorrectly based this off another branch

File: build-farm/platform-specific-configurations/linux.sh [hunk 2]
```diff
@@ -188,7 +191,7 @@ if [ "${VARIANT}" == "${BUILD_VARIANT_DRAGONWELL}" ] && [ "$JAVA_FEATURE_VERSION
   export CXX=/usr/local/gcc9/bin/g++-9.3
   # Enable GCC 10 for Java 18+ for repeatable builds, but not for our supported releases
```
[the remark this answers is not in the document]
Comment (sxa) [reply build-farm/platform-specific-configurations/linux.sh]: Thanks - fixed.

**Review by karianna — APPROVED** [review]
LGTM

**Review by andrew-m-leonard — APPROVED** [review]

**Review by keithc-ca — APPROVED** [review]
